### PR TITLE
Feature/1094/my entry sidebar

### PR DESCRIPTION
**Pull request by garyluu**
Changes to the my-entry side bar to have pubilsh/unpublish toggle for each organization accordion.

ga4gh/dockstore#1094

also has ga4gh/dockstore#1239

### CODE REVIEW

File: cypress/integration/mytools.js [hunk 1]
```diff
@@ -1,150 +1,154 @@
...
+                .click()
+
+            cy
+                .get('#deregisterButton')
+                .click()
+            cy
+                .get('#deregisterConfirmButton')
+                .click()
+
+            // This should be activated later
```
Comment (denis-yuen) [inline cypress/integration/mytools.js]: Maybe add what this is waiting for or link to a ticket

File: src/app/mytools/my-tool/my-tool.component.ts [hunk 2]
```diff
@@ -57,95 +60,192 @@ export class MyToolComponent implements OnInit {
...
+ * an array of published and unpublished workflows
+ * and which tab should be opened (published or unpublished)
+ * Main reason to convert to the new object is because figuring it out which tab should be active on
+ * the fly will result in function being executed far too many times (150 times)
+ * @private
+ * @param {Array<any>} nsWorkflows The original nsWorkflows object
+ * @returns {Array<OrgWorkflowObject>} The new object with more properties
+ * @memberof MyWorkflowComponent
+ */
+
```
Comment (denis-yuen) [inline src/app/mytools/my-tool/my-tool.component.ts]: Spacing intentional?

File: src/app/myworkflows/my-workflow/my-workflow.component.ts [hunk 3]
```diff
@@ -61,91 +65,214 @@ export class MyWorkflowComponent implements OnInit {
...
+  /**
+   * This figures out which tab (Published/Unpublished) is active
+   * In order of priority:
+   * 1. If the selected entry is published/unpublished, the tab selected will published/unpublished to reflect it
+   * 2. If there are published entries, the published tab will be selected
+   * 3. Unpublished otherwise
+   * @private
+   * @memberof MyWorkflowComponent
+   */
+  private updateActiveTab() {
```
Comment (denis-yuen) [inline src/app/myworkflows/my-workflow/my-workflow.component.ts]: Structure here seems like a copy-paste of https://github.com/dockstore/dockstore-ui2/pull/250/files#diff-bb18135c061ae38bef3adf1efab8d9dfR129 with only types changed.

File: src/app/mytools/my-tool/my-tool.component.ts [hunk 4]
```diff
@@ -57,95 +60,192 @@ export class MyToolComponent implements OnInit {
...
+   * In order of priority:
+   * 1. If the selected entry is published/unpublished, the tab selected will published/unpublished to reflect it
+   * 2. If there are published entries, the published tab will be selected
+   * 3. Unpublished otherwise
+   * @private
+   * @memberof MyToolComponent
+   */
+  private updateActiveTab() {
+    if (this.orgToolsObject) {
+      for (let i = 0; i < this.orgToolsObject.length; i++) {
```
Comment (denis-yuen) [inline src/app/mytools/my-tool/my-tool.component.ts]: Is it possible to clean up the array indices `[i]` with an iterator or similar ?
https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html

File: src/app/mytools/my-tool/my-tool.component.ts [hunk 5]
```diff
@@ -57,95 +60,192 @@ export class MyToolComponent implements OnInit {
...
+ * the fly will result in function being executed far too many times (150 times)
+ * @private
+ * @param {Array<any>} nsWorkflows The original nsWorkflows object
+ * @returns {Array<OrgWorkflowObject>} The new object with more properties
+ * @memberof MyWorkflowComponent
+ */
+
+  private convertNamespaceWorkflowsToOrgWorkflowsObject(nsWorkflows: Array<any>): Array<OrgToolObject> {
+    const orgWorkflowsObject: Array<OrgToolObject> = [];
+    for (let i = 0; i < nsWorkflows.length; i++) {
```
Comment (denis-yuen) [inline src/app/mytools/my-tool/my-tool.component.ts]: Same here

File: cypress/integration/mytools.js (hunk 1 above)
Comment (garyluu) [reply cypress/integration/mytools.js]: Most of these lines here got formatted, didn't intentionally add this

File: src/app/mytools/my-tool/my-tool.component.ts (hunk 4 above)
Comment (garyluu) [reply src/app/mytools/my-tool/my-tool.component.ts]: Ran into problems trying that, went with this guy's opinion instead https://stackoverflow.com/a/3010848/4484759

Comment (denis-yuen) [reply src/app/mytools/my-tool/my-tool.component.ts]: Gross. Boo@typescript

File: cypress/integration/mytools.js (hunk 1 above)
Comment (denis-yuen) [reply cypress/integration/mytools.js]: Still, no time like the present to figure this out

**Pull request merged**